pkg/minikube/cluster/cluster.go

```diff
@@ -228,11 +230,43 @@ var assets = []fileToCopy{
 	},
 }
 
-func UpdateCluster(d drivers.Driver) error {
+// Returns a function that will return n errors, then return successfully forever.
+func localkubeDownloader(resp *http.Response, config KubernetesConfig) func() error {
+	return func() (err error) {
+		tmpResp, err := http.Get(util.GetLocalkubeDownloadURL(config.KubernetesVersion,
+			constants.LocalkubeLinuxFilename))
+		resp.Body = tmpResp.Body
+		resp.ContentLength = tmpResp.ContentLength
+		return err
+	}
+}
+
+func UpdateCluster(h sshAble, d drivers.Driver, config KubernetesConfig) error {
 	client, err := sshutil.NewSSHClient(d)
 	if err != nil {
 		return err
 	}
+	if localkubeURLWasSpecified(config) {
```

Comment:
This might be clearer if you split these two statements into helper functions, then inline the localkubeDownloader. Something like:

func updateLocalkubeFromUrl(...) {

  inner := func() {
  ...
  }
  util.Retry(5, f)
}

func updateLocalkubeFromAsset(...) {

}